src/App.jsx

```diff
@@ -14,45 +15,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABETICALLY = 'Sort alphabetically';
+const SORT_FIELD_BY_LENGTH = 'Sort by length';
+
+const sortGoods = (goods, field, isReversed) => {
+  const sortedGoods = [...goods];
+
+  switch (field) {
+    case SORT_FIELD_ALPHABETICALLY:
+      sortedGoods.sort();
+      break;
+    case SORT_FIELD_BY_LENGTH:
+      sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const sortedGoods = sortGoods(goodsFromServer, sortField, isReversed);
+  const showResetButton = isReversed || sortField;
```

Comment:
```suggestion
  const isResetVisible = isReversed || sortField;
```
It should be clear from the name that the value is a boolean
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig